sdk/storage/azblob/zt_blob_client_test.go

```diff
@@ -2960,7 +2960,127 @@ func (s *azblobTestSuite) TestBlobSetMetadataIfNoneMatchFalse() {
 	validateBlobErrorCode(_require, err, bloberror.ConditionNotMet)
 }
 
-// nolint
+func (s *azblobTestSuite) TestPermanentDelete() {
+	_require := require.New(s.T())
```

Comment:
We shall also have negative test case there that with SAS not having permanent delete permission, if user tries permanent delete api it shall fail.